elastic/eui
# [Emotion] Convert `EuiKeyPadMenuItem`

## Summary

closes https://github.com/elastic/eui/issues/6653

As always, I recommend [following along by commit](https://github.com/elastic/eui/pull/7118/commits), as I included reasoning for various changes in commit messages, as well as misc cleanup items

### 🔲  Kibana follow-up items

- There's just [one class modifier usage](https://github.com/elastic/kibana/blob/99db840a8573442d6284439d845e664b0ac79087/x-pack/plugins/security/public/account_management/user_profile/user_profile.test.tsx#L394) being used in a test selector that will be need to be updated in the next upgrade. IMO, usage is low enough that it's worth nuking all selectors

## QA

- [x] Compare [staging](https://eui.elasit.co/pr_7118/#/navigation/key-pad-menu#menu-item) against [production](https://elastic.github.io/eui/#/navigation/key-pad-menu), and confirm the component looks the same as before
  - **NOTE:** There will likely be some minute typography differences due to Emotion containing more up to date font and line height style changes. It should not be major however.
- `gh checkout pr 7118`
- Go to http://localhost:6006/?path=/story/euikeypadmenuitem--playground
- [x] Play with various props (particularly permutations of `isSelected`, `isDisabled`, and `checkable` and confirm general UI/UX is working and non-broken

### General checklist


<details><summary><h3>Emotion checklist</h3></summary>

**Kibana usage**

  - ~[ ] If usage exists, consider converting to a `data` attribute so that consumers still have something to hook into~ - going to just look at the entire `className` string instead
- [x] Any test/query selectors that will need to be updated - **Yes, one**
- [x] Any Sass or CSS that will need to be updated, particularly if a component Sass var was deleted - **None**
- [x] Any direct className usages that will need to be refactored (e.g. someone calling the `euiBadge` class on a div instead of simply using the `EuiBadge` component) - **None**

---

**General**


---

**Unit tests**

    - Already added

---

**Sass/Emotion conversion process**

- [x] All SCSS overrides have been removed from the Amsterdam theme
- ~[ ] Any dependent components are identified in a new issue~

---

**CSS tech debt**


---

**DOM Cleanup**


---

**Extras/nice-to-have**

   - Note that this wasn't fully possible for all scenarios, e.g. when hovering a parent affects a child, or for affecting the fill color of SVG icons
</details>
